### PR TITLE
Dashboard Plugin

### PR DESCRIPTION
**Pull request by ritch**

### CODE REVIEW

File: plugins/custom-dashboard/__init__.py [hunk 1]
```diff
@@ -0,0 +1,913 @@
...
+
+        if isinstance(field, field_types) or (
+            isinstance(field, fo.ListField)
+            and isinstance(field.field, field_types)
+        ):
+            paths.append(path)
+
+    return paths
+
+def get_view_for_category(field, category, view):
```
Comment (brimoor) [inline plugins/custom-dashboard/__init__.py:100]: Okay this is a bit of a mouthful, but here's how to make this work for all possible types of fields the chart may be plotting:

```py
def get_view_for_value(sample_collection, path, value):
    """Returns a view into the given `sample_collection` that matches the given
    `value` within the given `path`.

    Supports label fields, list fields, and a combination of both.
    """
    root, leaf = _parse_path(sample_collection, path)
    is_label_field = _is_field_type(sample_collection, root, fo.Label)
    is_list_field = _is_field_type(sample_collection, path, fo.ListField)

    if is_label_field:
        if is_list_field:
            expr = F(leaf).exists() & F(leaf).contains(value)
        else:
            expr = F(leaf) == value

        return sample_collection.filter_labels(root, expr)

    if is_list_field:
        expr = F(path).exists() & F(path).contains(value)
    else:
        expr = F(path) == value

    return sample_collection.match(expr)


def _is_field_type(sample_collection, path, field_or_type):
    field = sample_collection.get_field(path)

    if issubclass(field_or_type, fo.Field):
        return isinstance(field, field_or_type)

    return (
        isinstance(field, fo.EmbeddedDocumentField)
        and issubclass(field.document_type, field_or_type)
    )


def _parse_path(sample_collection, path):
    if "." not in path:
        return path, None

    chunks = path.split(".")
    root = chunks[0]
    leaf = chunks[-1]

    # Handle dynamic documents
    idx = 0
    while (
        _is_field_type(sample_collection, root, fo.EmbeddedDocumentField)
        and not _is_field_type(sample_collection, root, fo.Label)
    ):
        idx += 1
        root += "." + chunks[idx]

    return root, leaf
```

Example usage:

```py
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart")

dataset[:100].tag_samples("hi")
dataset[:100].tag_labels("hi", label_fields="ground_truth")

view1 = get_view_for_value(dataset, "filepath", dataset.first().filepath)
view2 = get_view_for_value(dataset, "tags", "hi")
view3 = get_view_for_value(dataset, "ground_truth.detections.label", "cat")
view4 = get_view_for_value(dataset, "ground_truth.detections.tags", "hi")
```

File: plugins/custom-dashboard/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Custom Dashboard Plugin
```
Comment (brimoor) [inline plugins/custom-dashboard/README.md:1]: Can we rename `Custom Dashboard` to `Dashboard` for conciseness throughout here? This is *the* dashboarding tool 😄

File: plugins/custom-dashboard/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+A plugin that enables custom dashboards in the FiftyOne app.
```
Comment (brimoor) [inline plugins/custom-dashboard/README.md:3]: It's fine to say "custom dashboard" here, but I think the functional nomenclature should be just "dashboard".

**Review by brimoor — CHANGES_REQUESTED** [review]

File: plugins/custom-dashboard/__init__.py (hunk 1 above)
Comment (ritch) [reply plugins/custom-dashboard/__init__.py:100]: Clean.

File: plugins/custom-dashboard/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Dashboard Plugin
```
Comment (brimoor) [inline plugins/custom-dashboard/README.md:1]: @ritch can you rename the `custom-dashboard/` folder to `dashboard/`?

File: plugins/custom-dashboard/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+A plugin that enables dashboards of user defined plots in the FiftyOne app.
+
+https://github.com/user-attachments/assets/2ec5946e-ac43-48fa-b0a9-4d1d545ba0fc
+
+## Installation
+
+```shell
+fiftyone plugins download \
+    https://github.com/voxel51/fiftyone-plugins \
+    --plugin-names @voxel51/dashboard-builder
```
Comment (brimoor) [inline plugins/custom-dashboard/README.md:12]: Needs to be updated to match the final plugin name, which I propose is `@voxel51/dashboard`

File: plugins/custom-dashboard/fiftyone.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+name: "@voxel51/custom_dashboard"
```
Comment (brimoor) [inline plugins/custom-dashboard/fiftyone.yml:1]: Can we rename the plugin to `@voxel51/dashboard`?

File: plugins/custom-dashboard/fiftyone.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+description: Build custom dashboards for your FiftyOne datasets
+version: 1.0.0
+fiftyone:
+  version: ">=0.25.0"
+url: https://github.com/voxel51/fiftyone-plugins/tree/main/plugins/custom-dashboard
```
Comment (brimoor) [inline plugins/custom-dashboard/fiftyone.yml:6]: will need updating after folder name changes

**Review by brimoor — CHANGES_REQUESTED** [review]
@ritch can you also add an entry to the core plugins table? 
https://github.com/voxel51/fiftyone-plugins/blob/83ad567c39795819428028fbb66e5f86a72997d4/README.md?plain=1#L41

**Review by brimoor — APPROVED** [review]
LGTM 🚀

**Pull request merged**